Broadcasting error with sliced array and ParallelAccelerator

Running this:
```python
import numpy as np
from numba import jit

@jit(nopython=True, parallel=True)
def test(x1, x2):
    x1 = x1.reshape(x1.size, 1)
    x2 = x2.reshape(1, x2.size)
    return x1 >= x2[:, :-1]

x1 = np.random.rand(5)
x2 = np.random.rand(3)
test(x1, x2)
```
I get
```python
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-15-a9fd360d9c07> in <module>
     10 x1 = np.random.rand(5)
     11 x2 = np.random.rand(3)
---> 12 test(x1, x2)

AssertionError: Sizes of x1.1, $0.24 do not match on <ipython-input-15-a9fd360d9c07> (8)
```
using version 0.44.1. When setting parallel = False I get no error.

--- a/numba/array_analysis.py
+++ b/numba/array_analysis.py
@@ -1283,8 +1283,9 @@ class ArrayAnalysis(object):
             # Create a deepcopy of slice calltype so that when we change it below
             # the original isn't changed.  Make the types of the parts of the slice
             # intp.
-            self.calltypes[replacement_slice] = copy.deepcopy(self.calltypes[index_def])
-            self.calltypes[replacement_slice].args = (types.intp, types.intp)
+            new_arg_typs = (types.intp, types.intp)
+            rs_calltype = self.typemap[index_def.func.name].get_call_type(self.context, new_arg_typs, {})
+            self.calltypes[replacement_slice] = rs_calltype
             stmts.append(ir.Assign(value=replacement_slice, target=replacement_slice_var, loc=loc))
             # The type of the replacement slice is the same type as the original.
             self.typemap[replacement_slice_var.name] = self.typemap[index.name]

--- a/numba/tests/test_parfors.py
+++ b/numba/tests/test_parfors.py
@@ -2619,6 +2619,17 @@ class TestParforsSlice(TestParforsBase):
 
         self.check(test_impl, check_scheduling=False)
 
+    @skip_unsupported
+    def test_parfor_slice21(self):
+        def test_impl(x1, x2):
+            x1 = x1.reshape(x1.size, 1)
+            x2 = x2.reshape(x2.size, 1)
+            return x1 >= x2[:-1, :]
+
+        x1 = np.random.rand(5)
+        x2 = np.random.rand(6)
+        self.check(test_impl, x1, x2)
+
 class TestParforsOptions(TestParforsBase):
 
     def check(self, pyfunc, *args, **kwargs):

